pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -64,58 +67,81 @@ class Simulation:
 
             - A float to act as a sampling rate for the resulting state.
-        with_modulation: Whether to simulated the sequence with the programmed
-            input or the expected output.
     """
 
     def __init__(
         self,
-        sequence: Sequence,
+        sampled_seq: SequenceSamples,
+        register: BaseRegister,
+        device: BaseDevice,
         sampling_rate: float = 1.0,
         config: Optional[SimConfig] = None,
         evaluation_times: Union[float, str, ArrayLike] = "Full",
-        with_modulation: bool = False,
     ) -> None:
         """Instantiates a Simulation object."""
-        if not isinstance(sequence, Sequence):
+        # Initializing the samples obj
+        if not isinstance(sampled_seq, SequenceSamples):
             raise TypeError(
                 "The provided sequence has to be a valid "
-                "pulser.Sequence instance."
+                "SequenceSamples instance."
             )
-        if sequence.is_parametrized() or sequence.is_register_mappable():
+        if sampled_seq.max_duration == 0:
+            raise ValueError("SequenceSamples is empty.")
+        # Check compatibility of register and device
+        self.__device = device
+        self.__device.validate_register(register)
+        self.__register = register
+        # Check compatibility of samples and device:
+        if (
+            sampled_seq._slm_mask.end > 0
+            and not self.__device.supports_slm_mask
+        ):
             raise ValueError(
-                "The provided sequence needs to be built to be simulated. Call"
-                " `Sequence.build()` with the necessary parameters."
+                "Samples use SLM mask but device does not have one."
             )
-        if not sequence._schedule:
-            raise ValueError("The provided sequence has no declared channels.")
-        if all(
-            sequence._schedule[x][-1].tf == 0
-            for x in sequence.declared_channels
-        ):
+        if not sampled_seq.used_bases <= device.supported_bases:
             raise ValueError(
-                "No instructions given for the channels in the sequence."
+                "Bases used in samples should be supported by device."
             )
-        self._seq = sequence
-        self._interaction = "XY" if self._seq._in_xy else "ising"
-        self._qdict = self._seq.qubit_info
-        self._size = len(self._qdict)
-        self._modulated = with_modulation
-        if self._modulated and sequence._slm_mask_targets:
-            raise NotImplementedError(
-                "Simulation of sequences combining an SLM mask and output "
-                "modulation is not supported."
+        # Check compatibility of masked samples and register
+        if not sampled_seq._slm_mask.targets <= set(register.qubit_ids):
+            raise ValueError(
+                "The ids of qubits targeted in SLM mask"
+                " should be defined in register"
```

Comment:
nit: The error messages in this section could use a clean-up (punctuation and stuff like that)